packages/expo-ui/src/Button.types.ts

```diff
@@ -0,0 +1,77 @@
+import type { StyleProp, ViewStyle } from 'react-native';
+
+/**
+ * The role of the button.
+ * - `default` - The default button role.
+ * - `cancel` - A button that cancels the current operation.
+ * - `destructive` - A button that deletes data or performs a destructive action.
+ * @platform ios
+ */
+export type ButtonRole = 'default' | 'cancel' | 'destructive';
+
+/*
+ * The built-in button styles available on iOS and Android.
+ * Common styles:
+ * - `default` - The default system button style.
+ * - `bordered` - A button with a light fill. On Android equivalent to `FilledTonalButton`.
+ * - `borderless` - A button with no background or border. On Android equivalent to `TextButton`.
+ * Apple-only styles:
+ * - `borderedProminent` - A bordered button with a prominent appearance.
+ * - `plain` - A button with no border or background and a less prominent text.
+ * MacOS-only styles:
+ * - `accessoryBar` - A button style for accessory bars.
+ * - `accessoryBarAction` - A button style for accessory bar actions.
+ * - `card` - A button style for cards.
+ * - `link` - A button style for links.
+ * Android-only styles:
+ * - `outlined` - A button with an outline.
+ * - `elevated` - A filled button with a shadow.
+ *
+ * @platform android
+ * @platform ios
+ */
+export type ButtonStyle =
+  // Common
+  | 'default'
+  | 'bordered'
+  | 'plain'
+  // Apple-only
+  | 'borderedProminent'
+  | 'borderless'
+  // MacOS-only;
+  | 'accessoryBar'
+  | 'accessoryBarAction'
+  | 'card'
+  | 'link'
+  // Android-only;
+  | 'outlined'
+  | 'elevated';
+
+export type ButtonProps = {
+  /**
+   * The text to display inside the button.
+   */
+  text: string;
+  /**
+   * A callback that is called when the button is pressed.
+   */
+  onButtonPressed?: () => void;
+  /**
+   * A string describing the system image to display in the button.
+   * @platform ios
+   */
+  systemImage?: string;
```

Comment:
It'd be nice if there was also a native Material icons (not `expo/vector-icons`) in Jetpack. Devs could make a mapping in user-space that binds this (AI is pretty good at guessing for users). e.g. https://github.com/EvanBacon/expo-router-forms-components/blob/c360b4127549242c9ba0fe3e4030228b99a8c32a/components/ui/IconSymbolFallback.tsx#L10-L580

Reply:
https://developer.android.com/develop/ui/compose/graphics/images/material we may be able to